packages/server/src/server.ts

```diff
@@ -134,11 +135,48 @@ app.getKoaApp().use(async (ctx, next) => {
   }
 });
 
+app.getKoaApp().use(async (ctx, next) => {
```

Comment:
I feel like this section can be better formatted to read like 3 different properties being checked and potentially set in 3 different method with 1 method managing the orchestration of the 3.
Not a biggie for this PR but something to note for easier testing and reading

Reply:
That being said I don't see any tests for this logic.
Is it possible to test?